### ISSUE TITLE
two different Rust packages called `dpd-client`

### ISSUE THREAD

**Issue by davepacheco** [oxidecomputer/omicron#6666]
In our system we have two distinct packages called `dpd-client`:

- There's one in the "dendrite" repo.  This is used by:
    - `swadm` (in Dendrite)
    - `tfportd` (in Dendrite)`
    - `ddm` (in Maghemite)
    - `ddmd` (in Maghemite)
    - `mgd` (via `mg-lower`) (in Maghemite)
- There's one in the "omicron" repo.  This is used by:
    - `wicketd` (in Omicron)
    - `omicron-sled-agent` (in Omicron)
    - `omicron-nexus` (in Omicron)

Rust package names are supposed to be unique.  This happens to work, maybe because the packages in the above two groups are never built in the same workspace?  Based on this comment:
https://github.com/oxidecomputer/omicron/blob/68583ad7e931bfbab5f9b7c3aff9789609c84129/clients/dpd-client/build.rs#L5-L11

I gather that these clients are about equivalent and the second one exists only because the first one is private?  But Maghemite is also public but still seems to use the (private) Dendrite client.

The new `ls-apis` tool works around this:
https://github.com/oxidecomputer/omicron/blob/68583ad7e931bfbab5f9b7c3aff9789609c84129/dev-tools/ls-apis/src/workspaces.rs#L157-L195

**Comment by davepacheco** [comment]
@Nieuwejaar is this something we can resolve now?

**Comment by Nieuwejaar** [comment]
Yes.  This should be completed by d6f974eee13f3abafef3526c5d948655f81f9105.

Thanks for noticing this and calling it out.
